src/Apps/Legion/Components/WizardShell/Internal/ModifyStep/Internal/TypesTab/Internal/TypesCard/TypesCard.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useMemo } from 'react';
+import { getDebugLogger } from '../../../../../../../../../../Models/Services/Utils';
+import { ITypesCardProps } from './TypesCard.types';
+import { getStyles } from './TypesCard.styles';
+import { useTypes } from '../../../../../../../../Hooks/useTypes';
+import {
+    DetailsList,
+    IColumn,
+    TextField,
+    SelectionMode,
+    Checkbox
+} from '@fluentui/react';
+import { useEntities } from '../../../../../../../../Hooks/useEntities';
+import TypeIcon from '../../../../../../../TypeIcon/TypeIcon';
+import { IViewProperty } from '../../../../../../../../Models';
+import { useExtendedTheme } from '../../../../../../../../../../Models/Hooks/useExtendedTheme';
+import { useTranslation } from 'react-i18next';
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('TypesCard', debugLogging);
+
+const getColumns = (): IColumn[] => [
```

Comment:
could use the `memoizeFunction` helper here to avoid rerendering the details list on every render of this component.